### PR TITLE
fix: return partition table not exist when trying to read an empty dev

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/go-blockdevice/issues/51

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

**Comment by Unix4ever** [comment]
talos side https://github.com/talos-systems/talos/pull/4566

File: blockdevice/options.go [hunk 1]
```diff
@@ -35,10 +36,18 @@ func WithMode(value int) Option {
...
+	return func(args *Options) {
+		args.IgnoreGPT = value
+	}
+}
+
 // NewDefaultOptions initializes a Options struct with default values.
 func NewDefaultOptions(setters ...Option) *Options {
 	opts := &Options{
 		CreateGPT: false,
+		IgnoreGPT: false,
```
Comment (smira) [inline blockdevice/options.go:50]: I don't see so far why do we need this flag. 

It looks like `IgnoreGPT` is something we always had before - if the GPT partition is missing, `Open` still worked, but partition table was nil.

Should we just treat GPT header failures same way as missing PMBR?

**Review by smira — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]

**Pull request merged**